server/controllers/receipt-controller.js

```diff
@@ -32,16 +32,27 @@ const createReceipt = async (req, res, next) => {
 // @access Private
 const updateReceipt = async (req, res, next) => {
     try {
+        // check
+        const receipt = await Receipt.findById(req.params.id);
+        if (!receipt) {
+            const error = new HttpError('Receipt not found', 404);
+            return next(error);
+        }
+        if (receipt.user != req.userData.userId) {
+            const error = new HttpError('Not Authorized', 401);
+            return next(error);
+        }
```

Comment:
what are you getting in receipt  const is a receipt object  and in line: 41 you are comparing receipt object with userId.

Reply:
receipit.user contains the user object id, thats for checking if the receipt belongs to the user